libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -593,7 +604,9 @@ internal class MapboxTripSession(
         shouldTriggerBannerInstructionsObserver: Boolean
     ) {
         routeProgress = progress
-        tripService.updateNotification(buildTripNotificationState(progress))
+        if (withTripService) {
```

Comment:
```suggestion
        if (tripService.hasServiceStarted()) {
```

Reply:
This would still be `if(withTripService)` because you need to hook on to the parameter the user has passed to decide whether to start the service or not. Not hook on to the current state of the service which would always return false if hasn't been started ever and you want to start it with the session.